core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -6660,6 +6661,61 @@ class ReplicaManagerTest {
     }
   }
 
+  @Test
+  def testDeleteRecordsInternalTopicDeleteDisallowed(): Unit = {
```

Comment:
There are lots of existing usage of `rm.becomeLeaderOrFollower`. It would be useful to clean them up in a followup jira.

Reply:
@junrao 
I would expect the code owners to handle that.